### PR TITLE
add test for Publisher and Subscriber snippets

### PR DESCRIPTION
**Pull request by pongad**
The test code isn't as clear as I hoped, but I'm not sure how to simplify it without complicating the samples.

FYI @jabubake

### CODE REVIEW

**Comment by lesv** [comment]
Of course, you need to fix what ever is making Travis unhappy.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java [hunk 1]
```diff
@@ -264,4 +277,79 @@ public void testTopicPolicyAsync() throws ExecutionException, InterruptedExcepti
...
+        new SubscriberSnippets(
+            subscriptionName,
+            new MessageReceiverSnippets(queue).messageReceiver(),
+            done,
+            MoreExecutors.directExecutor());
+    new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          received.set(queue.poll(10, TimeUnit.MINUTES));
```
Comment (lesv) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java]: That's a long time to wait in travis.

**Comment by pongad** [comment]
@lesv The Travis failure seems to be coming from Spanner and unrelated to this change. Nearest I can tell, it might be from temporary connection issue while downloading Maven packages. I reran to make sure.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java (hunk 1 above)
Comment (pongad) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java]: I agree that this is a long time to wait. On the other hand, if this times out, the whole test fails anyway right?

**Comment by lesv** [comment]
LGTM from me, ping to @garrettjonesgoogle and @jabubake

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**